tools/publish-aos.js

```diff
@@ -1,54 +1,170 @@
 const Arweave = require('arweave');
 const path = require('path');
 const fs = require('fs');
+const { TurboFactory } = require('@ardrive/turbo-sdk');
+const { createData, ArweaveSigner } = require('@dha-team/arbundles');
+
+const changelog = `
+# Changelog
+
+### Changed
+
+- Changed default landing page transaction ID.
+`;
 
 const srcCodeTxIdPlaceholder = '__INSERT_SOURCE_CODE_ID__';
 
 const arweave = Arweave.init({
   host: 'arweave.net',
   port: 443,
   protocol: 'https',
+  logging: false,
 });
+
+function getPublishedSourceCodeIds() {
+  // Read all filenames from the changelogs directory
+  const changeLogFileNames = fs.readdirSync(
+    path.join(__dirname, '../changelogs'),
+  );
+
+  // Process each file name to extract the source code ID
+  const txIds = changeLogFileNames
+    .map((fileName) => {
+      // Remove the date portion at the start, which is in the format YYYY-MM-DD-
+      const parts = fileName.split('-').slice(3); // Skip the first 3 parts (YYYY, MM, DD)
+      const idWithExtension = parts.join('-'); // Join the remaining parts in case the ID itself has dashes
+      const txId = idWithExtension.split('.').shift(); // Remove the file extension (e.g., .md)
+
+      return txId;
+    })
+    .filter((txId) => txId.length === 43); // Filter for valid IDs with length 43
+
+  return txIds;
+}
 async function main() {
+  // get the last published source code from the changelogs folder (can check the date on it, check for multiple dates) and
+  // fetch the code from arweave, then compare it with the current code
+
+  const publishedSourceCodeIds = getPublishedSourceCodeIds();
+  const sourceCodeTransactions = await arweave.api
+    .post('/graphql', {
+      query: `{transactions(
+        ids:[ ${publishedSourceCodeIds.map((id) => `"${id}"`).join(',')}]) {
+			edges {
+				node {
+					id
+					tags {
+						name
+						value
+					}
+          block {
+            timestamp
+          }
+				}
+			}
+		}
+        }`,
+    })
+
+    .then((res) => {
+      return res.data.data.transactions.edges
+        .map((edge) => edge.node)
+        .sort((a, b) => b.block.timestamp - a.block.timestamp);
+    });
+
+  const lastPublishedSourceCodeId = sourceCodeTransactions[0].tags.find(
+    (tag) => tag.name === 'Original-Tx-Id',
+  )?.value;
+
+  const lastPublishedSourceCode = await arweave.transactions
+    .getData(lastPublishedSourceCodeId, { decode: true, string: true })
+    .catch((error) => {
+      console.error(
+        'Error fetching last published source code, unable to compare with current code. \n',
+        error,
+      );
+    });
+
   const bundledLua = fs.readFileSync(
     path.join(__dirname, '../dist/aos-bundled.lua'),
     'utf-8',
   );
-  const wallet = fs.readFileSync(path.join(__dirname, 'key.json'), 'utf-8');
+
+  if (lastPublishedSourceCode === bundledLua) {
```

Comment:
hmm, wondering if we should do a hash check here and add that to the tags on publishing